src/utils.c

```diff
@@ -99,14 +99,17 @@ SEXP allNAR(SEXP x) {
  *   existing columns for character
  *   optionally check for no duplicates
  */
-SEXP colnamesInt(SEXP x, SEXP cols, SEXP check_dups) {
+SEXP colnamesInt(SEXP x, SEXP cols, SEXP check_dups, SEXP skip_absent) {
   if (!isNewList(x))
     error(_("'x' argument must be data.table compatible"));
   if (!IS_TRUE_OR_FALSE(check_dups))
     error(_("%s must be TRUE or FALSE"), "check_dups");
+  if (!IS_TRUE_OR_FALSE(skip_absent))
+    error(_("%s must be TRUE or FALSE"), "skip_absent");
   int protecti = 0;
   R_len_t nx = length(x);
   R_len_t nc = length(cols);
+  bool Skip_absent = LOGICAL(skip_absent)[0];
```

Comment:
nit: naming style here is not consistent with our codebase. That said, we have not been particularly consistent ourselves. Here's some similar entries I find in src/ now:

```
src/nafill.c:  bool binplace = LOGICAL(inplace)[0];
src/nafill.c:  bool nan_is_na = LOGICAL(nan_is_na_arg)[0];
src/frollR.c:  bool badaptive = LOGICAL(adaptive)[0];
src/frollR.c:  bool bnarm = LOGICAL(narm)[0];
src/fcast.c:  bool some_fill = LOGICAL(some_fillArg)[0];
src/uniqlist.c:  bool narm = LOGICAL(narmArg)[0]==1;
src/transpose.c:  bool ignore = LOGICAL(ignoreArg)[0];
src/transpose.c:  bool listCol = LOGICAL(listColsArg)[0];
```

It looks like the most common choice is `skip_absent_arg` and then `bool skip_absent = LOGICAL(skip_absent_arg)[0]`.

Reply:
OTOH we have `check_dups` here, `skip_absent_arg` is a bit inconsistent with that.

Let's go with `bool bskip_absent = LOGICAL(skip_absent)[0]`. for this PR. Maybe we can revisit this naming style generally in the codebase.